Raven.Client.Lightweight/Document/RavenClientEnlistment.cs

```diff
@@ -63,13 +62,13 @@ public void Prepare(PreparingEnlistment preparingEnlistment)
 
 			    if (recoveryInformation == null)
 			    {
-			        session.PrepareTransaction(transaction.LocalIdentifier);
+					AsyncPump.Run(() => session.PrepareTransaction(transaction.LocalIdentifier));
```

Comment:
you can use AsyncHelpers for this instead of introducing AsyncPump:
https://github.com/ravendb/ravendb/blob/master/Raven.Abstractions/Util/AsyncHelpers.cs